### ISSUE TITLE
Local InfluxDB (Linux) and Containerized Kapacitor (Docker) Not Communicating

### ISSUE THREAD

**Issue by mingraham** [influxdata/kapacitor#580]
Hello,

I've spent the past couple of days trying to figure this out (surfing through some of the issues and Googling much), so I'm going to just ask now. I am dockerizing the TICK stack (or a version of it) at my job, and I have the case where each component is in its own container already completed (i.e., influxdb in its own container and kapacitor in its own container).

Where I am stuck is when there is an InfluxDB (say in production, which is why I am so concerned with this use case), and I want to bring in visualization (we are using Grafana, and this portion is working) and alerting via dockerized solutions, meaning we do not want to locally install kapacitor next to the influxdb. When I `docker-compose up -d` on my `docker-compose.yml` file, the kapacitor service successfully creates subscriptions in the InfluxDB. So this is great, because it means kapacitor was able to access influxdb from container to host.

The issue I am running into is that there seems to be no data being transmitted from InfluxDB to Kapacitor, as if it doesn't know how to get the data into the container. When I hop into the container and `kapacitor show cpu` (i named my alert cpu), it is enabled with no errors, though there is no data processed on the edges, meaning it has received no data, right?

Furthermore, when i `sudo tcpdump -n udp port 12345 -vv` (where once the port was 12345), nothing is happening on that port.

Any help is appreciated, thank you.

**Comment by nathanielc** [comment]
@mingraham To make sure I understand.

1. InfluxDB is running on different host than Kapacitor
2. Both processes are running inside docker containers

Correct?

If so a couple of questions:

1. What is the `hostname` setting in the Kapacitor config. If you are in the InfluxDB container will that hostname resolve to the correct address of the Kapacitor host/container?
2. Have you opened up the UDP ports on both the Kapacitor host and the container?
3. You can try sending UDP packets manually using `netcat` or similar tool from the InfluxDB container to the Kapacitor container to test network connectivity.

A few comments:

1. You are not the first to have significant pain in setting up subscriptions for Kapacitor in Docker. As such we are updating the subscriptions to be able to also use HTTP so its less port/network management to do. See https://github.com/influxdata/kapacitor/pull/562 For details
2. Kapacitor will honor whatever subscriptions already exist so if you need to define the ports specifically you can create the subscriptions manually and Kapacitor will just use them. Just make sure the subscription name matches the name when Kapacitor creates them automatically.

**Comment by mingraham** [comment]
Hi, thank you for getting back so quickly.

InfluxDB is NOT running inside a container, it is sitting on the Linux host. Kapacitor is dockerized ON that Linux host, so, to my understanding, I should just have to map ports from the container to the host it is sitting on, which, again to my understanding, should make it easy to get InfluxDB to be able to send data into the dockerized Kapacitor.

In the Kapacitor config file, I have tried many things, but I tried setting the hostname to `localhost` most recently, since I saw in one of the other issues that the hostname in the config should match the destination of the subscription in InfluxDB.

I, most recently, attempted opening up UDP ports from Kapacitor's container to InfluxDB on the host, and then manually add subscriptions in InfluxDB pointed to those ports with `http://localhost:<port>`, but had no such luck. Maybe I did that part incorrectly?

I won't try your 3rd question, since we clarified Kapacitor was in a container and InfluxDB was not, and you might have other suggestions.

That is great to hear about the HTTP subscriptions, and being able to manually subscribe to Kapacitor is nice, though since I am heavily automating this process, it isn't scalable. My goal is to be able to have maximum portability between environments with the exact same TICK stack, distinguishing the environments by environment variables in each environment. This is the last part I need to finish it, so thanks so much for the quick response, again.
